egg/disk_management/disk.py

```diff
@@ -0,0 +1,34 @@
+import parted
+from parted import Device
+
+from egg.disk_management.partition import Partition
+
+
+class Disk(object):
+    def __init__(self, device: Device = None, device_path: str = None) -> None:
+        if device is not None:
+            self.device: parted.Device = device
+        elif device_path:
+            self.device: parted.Device = parted.getDevice(device_path)
+        else:
+            raise parted.DiskException("no device specified")
+        self.disk: parted.Disk = parted.newDisk(self.device)
+        self.partitions: list = list(map(lambda x: Partition(x), self.disk.partitions))
+
+    def __del__(self) -> None:
+        self.device.removeFromCache()
+
+    def get_model(self) -> str:
+        return self.device.model.__str__()
```

Comment:
```suggestion
        return str(self.device.model)
```

Reply:
Can you just explain to me why I shouldn't use the attribute directly with double underscore?

Reply:
`__str__` is a "magic method" that can be implemented to create a "pretty" string representation of an object (just like `__add__` can be implemented to overload the `+` operator).
Python also supports `__repr__`  to create a "full" string representation of the object.
It is thus a customization point for implementors, but is not necessarily meant to be called explicitly.

On the other hand, `str(...)` is simply the way to construct an instance of the `str` class from an object, and takes care of choosing the preferred / available conversion method for us.
Thus, the dirty details of the conversion are abstracted away from our code.